packages/babel-parser/test/fixtures/core/categorized/for-async-of-dot/input.js

```diff
@@ -0,0 +1 @@
+for (async.x of y);
```

Comment:
Can you add a new test case when async contains escape?
```js
for (\u0061sync of []);
```
It should parsed successfully.

Note that currently V8 throws, I think this is a V8 bug.

Reply:
There's a test for that in the generator suite, but I can add one to the parser suite as well.